promql/parser/parse.go

```diff
@@ -388,7 +388,11 @@ func (p *parser) Lex(lval *yySymType) int {
 		lval.item.Typ = EOF
 		p.InjectItem(0)
 	case RIGHT_PAREN:
-		p.closingParens = append(p.closingParens, lval.item.Pos+posrange.Pos(len(lval.item.Val)))
+		if len(p.closingParens) >= 2 {
```

Comment:
I *think* this is only needed because there might be some cases where the closingParens isn't consumed and this code here hides that fact. But figuring out where it *should* be consumed isn't obvious to me.